### PR TITLE
reat: retry yarn installs and git clone

### PR DESCRIPTION
**Pull request by mshanemc**
retries around yarn install since it sometimes timeout/503
https://github.com/forcedotcom/source-deploy-retrieve/actions/runs/4602154013/jobs/8130861374

git clone will sometimes 429
https://github.com/salesforcecli/plugin-source/actions/runs/4621491649/jobs/8173050988?pr=800

### CODE REVIEW

File: .github/workflows/externalNut.yml [hunk 1]
```diff
@@ -84,7 +84,14 @@ jobs:
           max_attempts: ${{ inputs.attempts }}
           command: npm install -g sfdx-cli@latest-rc @salesforce/cli@latest-rc shx yarn-deduplicate --omit=dev
           timeout_minutes: 20
-      - run: git clone -b ${{ inputs.branch}} --single-branch ${{ inputs.externalProjectGitUrl}} $(pwd)
+      - uses: nick-fields/retry@943e742917ac94714d2f408a0e8320f2d1fcafcd
+        name: git clone
+        with:
+          max_attempts: ${{ inputs.attempts }}
```
Comment (iowillhoit) [inline .github/workflows/externalNut.yml:90]: You use `input.attempts` here but not below, was that intentional?

**Comment by iowillhoit** [comment]
A few thoughts: 
- It is possible that the 10 minute timeout would be hit on a Windows install after a retry or two. Maybe increase it a bit. 
- Instead of repeating this block, we could move the retry to a reusable workflow. Then it would remain a single line
  - Example: `- uses: salesforcecli/github-workflows/.github/workflows/yarn-install-with-retries.yml@main`

File: .github/workflows/externalNut.yml (hunk 1 above)
Comment (mshanemc) [reply .github/workflows/externalNut.yml:90]: seemed like a fine number to me.

**Comment by mshanemc** [comment]
> A few thoughts:
> 
> * It is possible that the 10 minute timeout would be hit on a Windows install after a retry or two. Maybe increase it a bit.

> * Instead of repeating this block, we could move the retry to a reusable workflow. Then it would remain a single line
>   
>   * Example: `- uses: salesforcecli/github-workflows/.github/workflows/yarn-install-with-retries.yml@main`

File: .github/workflows/externalNut.yml [hunk 2]
```diff
@@ -84,7 +84,14 @@ jobs:
           max_attempts: ${{ inputs.attempts }}
           command: npm install -g sfdx-cli@latest-rc @salesforce/cli@latest-rc shx yarn-deduplicate --omit=dev
           timeout_minutes: 20
-      - run: git clone -b ${{ inputs.branch}} --single-branch ${{ inputs.externalProjectGitUrl}} $(pwd)
+      - uses: nick-fields/retry@943e742917ac94714d2f408a0e8320f2d1fcafcd
+        name: git clone
+        with:
+          max_attempts: 20
+          command: git clone -b ${{ inputs.branch}} --single-branch ${{ inputs.externalProjectGitUrl}} $(pwd)
+          # backoff 60 seconds because most failures are 429 (too many requests)
```
Comment (iowillhoit) [inline .github/workflows/externalNut.yml:92]: Good call 👍

**Review by iowillhoit — APPROVED** [review]

**Pull request merged**